harmonic/install_ubuntu_src.md

```diff
@@ -0,0 +1,279 @@
+
+```bash
+colcon build --merge-install
+```
+
+To speed up the build process, you could also disable tests by using
+
+```bash
+colcon build --cmake-args -DBUILD_TESTING=OFF --merge-install
+```
+
+To use debuggers activate debug symbols. Gazebo will run slower, but you'll be able to use GDB:
+
+```bash
+colcon build --cmake-args ' -DBUILD_TESTING=OFF' ' -DCMAKE_BUILD_TYPE=Debug' --merge-install
+```
+
+To build a specific package with all its dependent packages:
+
+```bash
+colcon build --merge-install --packages-up-to PACKAGE_NAME
+```
+
+To build a single package:
+
+```bash
+colcon build --packages-select PACKAGE_NAME
+```
+
+Visit [colcon documentation](https://colcon.readthedocs.io/en/released/#) to view more `colcon` build and test options.
+
+If there are no errors, all the binaries should be ready to use. You can check the [Troubleshooting] section for errors.
+
+## Using the workspace
+
+The workspace needs to be sourced every time a new terminal is used.
+
+Run the following command to source the workspace in bash:
+
+```bash
+. ~/workspace/install/setup.bash
+```
+
+Or in zsh:
+
+```zsh
+. ~/workspace/install/setup.zsh
+```
+
+This is the end of the source install instructions; head back to the [Getting started](/docs/all/getstarted)
+page to start using Gazebo!
+
+## Uninstalling source-based install
+
+A source-based install can be "uninstalled" using several methods, depending on
+the results you want:
+
+  1. If you installed your workspace with `colcon` as instructed above, "uninstalling"
+     could be just a matter of opening a new terminal and not sourcing the
+     workspace's `setup.sh`. This way, your environment will behave as though
+     Gazebo is not installed on your system.
+
+  2. If, in addition to not wanting to use the libraries, you're also trying to
+     free up space, you can delete the entire workspace directory with:
+
+     ```bash
+     rm -rf ~/workspace
+     ```
+
+  3. If you want to keep the source code, you can remove the
+     `install` / `build` / `log` directories as desired, leaving the `src` directory.
+
+## Troubleshooting
+
+See [Troubleshooting](/docs/Harmonic/troubleshooting#ubuntu)
+
+## QML Debugging
+
+To perform QML debugging you'll need:
+
+ - Add `--cmake-args -DDQT_QML_DEBUG` flag to colcon
+ - QtCreator
+
+You will need to build Gazebo with:
+
+```bash
+colcon build --cmake-args -DQT_QML_DEBUG --merge-install
+```
+
+> **Note:** Advanced users may note that only the `gz-sim` project needs this flag.
+
+After that's done, launching `gz sim -g` will result in the following message:
+
+```
+QML debugging is enabled. Only use this in a safe environment.
+QML Debugger: Waiting for connection on port 40000...
+```
+
+After that you can just head to
+`QtCreator -> Debug -> Start Debugging -> Attach to QML Port...`
+and enter the QML port
+
+![](images/GzGuiQmlDebugging01.png)
+
+Once you click there, set the port number to 40000 and hit ok
+
+![](images/GzGuiQmlDebugging02.png)
```

Comment:
where is this image?

Reply:
added [9a931fd](https://github.com/gazebosim/docs/pull/324/commits/9a931fd599b1f5184266b22a1150090b837eae07)